packages/nextjs/hooks/scaffold-eth/useScaffoldEventRead.ts

```diff
@@ -0,0 +1,124 @@
+import { ContractAbi, ContractName } from "~~/utils/scaffold-eth/contract";
+import { ExtractAbiEventNames } from "abitype";
+import { useProvider, useContract } from "wagmi";
+import { useDeployedContractInfo } from "~~/hooks/scaffold-eth";
+import { ethers } from "ethers";
+import { useEffect, useState } from 'react';
+
+/**
+ * @dev reads events from a deployed contract
+ * @param config - The config settings
+ * @param config.contractName - deployed contract name
+ * @param config.eventName - name of the event to listen for
+ * @param config.fromBlock - the block number to start reading events from (default: 1000 blocks before current block)
+ * @param config.filters - filters to be applied to the event (parameterName: value)
+ * @param config.blockData - if set to true it will return the block data for each event (default: false)
+ * @param config.transactionData - if set to true it will return the transaction data for each event (default: false)
+ * @param config.receiptData - if set to true it will return the receipt data for each event (default: false)
+ */
+export const useScaffoldEventRead = <
+  TContractName extends ContractName,
+  TEventName extends ExtractAbiEventNames<ContractAbi<TContractName>>,
+>({
+  contractName,
+  eventName,
+  fromBlock,
+  filters,
+  blockData,
+  transactionData,
+  receiptData,
+}: {
+  contractName: TContractName;
+  eventName: TEventName;
+  fromBlock?: number;
+  filters?: any;
+  blockData?: boolean;
+  transactionData?: boolean;
+  receiptData?: boolean;
+}) => {
+  const [events, setEvents] = useState(<any>[]);
+  const { data: deployedContractData } = useDeployedContractInfo(contractName);
+  const provider = useProvider();
+
+  const contract = useContract({
+    address: deployedContractData?.address,
+    abi: deployedContractData?.abi,
+    signerOrProvider: provider,
+  });
+
+  useEffect(() => {
+    async function readEvents() {
+      if (!contract) {
+        console.log("contract not found");
+        return;
+      }
+      try {
+        if (!fromBlock) {
+          fromBlock = await provider.getBlockNumber() - 1000;
+        }
+
+        const fragment = contract.interface.getEvent(eventName);
+        if (!fragment) {
```

Comment:
Since our `eventName` is strongly typed to only allow valid event names, the fragment should always be defined, right? Maybe this condition can be improved a bit (or omitted by asserting its type?).

Reply:
Yes! You are right! What do you mean by "or omitted by asserting its type?"?

Reply:
Removed! It was throwing an exception anyway, so the catch block can catch it if it's needed.

Reply:
> What do you mean by "or omitted by asserting its type?

Hadn't opened the PR in vscode yet, so wasn't sure if typescript would complain about fragment possibly being undefined or not. Guess it wasn't an issue, based on your most recent changes, nice :+1: